### ISSUE TITLE
Difference between jcqp and qpoases?

### ISSUE THREAD

**Issue by littleggghost** [mit-biomimetics/Cheetah-Software#12]
Hi, there
I noticed that recent commit can use jcqp as the QP slover to replace qpoases, so my question is: Is jcqp more faster than qpoases? Or other reason?
Thanks in advance!

**Comment by dicarlo236** [comment]
By default, it should still be using qpOASES.  The controller's settings here (https://github.com/mit-biomimetics/Cheetah-Software/blob/master/config/mc-mit-ctrl-user-parameters.yaml) are set up for running qpOASES.

We added the option for JCQP and OSQP (which both implement a very similar algorithm), and a different sparse formulation of the MPC optimization, but we found that this algorithm had accuracy issues with the default settings.  If we tuned the solver settings to fix the accuracy issue, JCQP and OSQP were both much slower than qpOASES. In the end, qpOASES still seems to be the best for our problem.

In the most recent commit, we run the controller at 500 Hz instead of 1 kHz, which was required to run the new whole body + MPC controller.  We haven't seen any loss of performance when running at 500 Hz.
